src/nbl/video/CVulkanPhysicalDevice.h

```diff
@@ -1,88 +1,208 @@
+            m_features.robustBufferAccess = features.robustBufferAccess;
+            m_features.imageCubeArray = features.imageCubeArray;
+            m_features.logicOp = features.logicOp;
+            m_features.multiDrawIndirect = features.multiDrawIndirect;
+            m_features.multiViewport = features.multiViewport;
+            m_features.vertexAttributeDouble = features.shaderFloat64;
+            m_features.dispatchBase = false; // Todo(achal): Umm.. what is this?
+            m_features.shaderSubgroupBasic = subgroupProperties.supportedOperations & VK_SUBGROUP_FEATURE_BASIC_BIT;
+            m_features.shaderSubgroupVote = subgroupProperties.supportedOperations & VK_SUBGROUP_FEATURE_VOTE_BIT;
+            m_features.shaderSubgroupArithmetic = subgroupProperties.supportedOperations & VK_SUBGROUP_FEATURE_ARITHMETIC_BIT;
+            m_features.shaderSubgroupBallot = subgroupProperties.supportedOperations & VK_SUBGROUP_FEATURE_BALLOT_BIT;
+            m_features.shaderSubgroupShuffle = subgroupProperties.supportedOperations & VK_SUBGROUP_FEATURE_SHUFFLE_BIT;
+            m_features.shaderSubgroupShuffleRelative = subgroupProperties.supportedOperations & VK_SUBGROUP_FEATURE_SHUFFLE_RELATIVE_BIT;
+            m_features.shaderSubgroupClustered = subgroupProperties.supportedOperations & VK_SUBGROUP_FEATURE_CLUSTERED_BIT;
+            m_features.shaderSubgroupQuad = subgroupProperties.supportedOperations & VK_SUBGROUP_FEATURE_QUAD_BIT;
+            m_features.shaderSubgroupQuadAllStages = ((subgroupProperties.supportedStages & asset::ISpecializedShader::E_SHADER_STAGE::ESS_ALL)
+                                                        == asset::ISpecializedShader::E_SHADER_STAGE::ESS_ALL);
+        }
+                
         uint32_t qfamCount = 0u;
         vkGetPhysicalDeviceQueueFamilyProperties(m_vkphysdev, &qfamCount, nullptr);
         core::vector<VkQueueFamilyProperties> qfamprops(qfamCount);
         vkGetPhysicalDeviceQueueFamilyProperties(m_vkphysdev, &qfamCount, qfamprops.data());
-
+                
         m_qfamProperties = core::make_refctd_dynamic_array<qfam_props_array_t>(qfamCount);
         for (uint32_t i = 0u; i < qfamCount; ++i)
         {
             const auto& vkqf = qfamprops[i];
             auto& qf = (*m_qfamProperties)[i];
-
+                    
             qf.queueCount = vkqf.queueCount;
             qf.queueFlags = static_cast<E_QUEUE_FLAGS>(vkqf.queueFlags);
             qf.timestampValidBits = vkqf.timestampValidBits;
             qf.minImageTransferGranularity = { vkqf.minImageTransferGranularity.width, vkqf.minImageTransferGranularity.height, vkqf.minImageTransferGranularity.depth };
         }
     }
-
+            
     inline VkPhysicalDevice getInternalObject() const { return m_vkphysdev; }
-
+            
     E_API_TYPE getAPIType() const override { return EAT_VULKAN; }
 
+    std::vector<ISurface::SFormat> getAvailableFormatsForSurface(const ISurface* surface) const override
```

Comment:
do not return a `vector` container, have the API similar to `vkGetPhysicalDeviceSurfaceFormatsKHR` pointer and a count reference